src/core/view/StrokeViewHelper.cpp

```diff
@@ -49,9 +45,9 @@ double xoj::view::StrokeViewHelper::drawWithPressure(cairo_t* cr, const std::vec
         cairo_stroke(cr);
     };
 
-    if (dashes) {
+    if (!dashes.empty()) {
         for (const auto& [p, q]: PairView(pts)) {
-            cairo_set_dash(cr, dashes, dashCount, dashOffset);
+            cairo_set_dash(cr, dashes.data(), static_cast<int>(dashes.size()), dashOffset);
```

Comment:
We could have a helper function, either LineStyle::set_cairo_dashes or Util::cairo_set_dash_from_vector. This would avoid the code duplication